### PR TITLE
tracker: implement graceful

### PR DESCRIPTION
**Pull request by kipukun**
WIP

### CODE REVIEW

**Comment by kipukun** [comment]
refers to issue #115

File: tracker/server.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+	recorder *Recorder
+	http     *http.Server
+	grpc     *grpc.Server
+	httpLn   net.Listener
+	grpcLn   net.Listener
+
+	h http.Handler
+}
+
+func (s *Server) Start(ctx context.Context, fds *fdstore.Store) error {
```
Comment (Wessie) [inline tracker/server.go:51]: Can move this functions content back into execute and use the `httpLn` data slot for storing the recorder state

File: tracker/graceful.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		return err
+	}
+	err = fds.AddListener(s.grpcLn, GrpcLn, nil)
+	if err != nil {
+		return err
+	}
+	return s.recorder.storeSelf(ctx, fds)
+}
+
+func (r *Recorder) storeSelf(ctx context.Context, fds *fdstore.Store) error {
```
Comment (Wessie) [inline tracker/graceful.go:28]: Either just make these generic `UnmarshalJSON`/`MarshalJSON` functions on the recorder and call them from the server store function or use the protobuf types in `rpc`, either works

File: tracker/graceful.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	radio "github.com/R-a-dio/valkyrie"
+	"github.com/Wessie/fdstore"
+)
+
+type wireRecorder struct {
+	Listeners map[radio.ListenerClientID]*Listener
+}
+
+func (s *Server) storeSelf(ctx context.Context, fds *fdstore.Store) error {
+	err := fds.AddListener(s.httpLn, HttpLn, nil)
```
Comment (Wessie) [inline tracker/graceful.go:17]: use this data slot for the recorder state, you can marshal it to bytes right above it

**Pull request merged**